components/derive/src/lib.rs

```diff
@@ -249,11 +249,43 @@ pub fn service_factory(input: TokenStream) -> TokenStream {
 ///
 /// Prefix of the `exonum` crate has two main values - `crate` or `exonum`. The default value
 /// is `exonum`.
+///
+/// # Method attributes
+///
+/// ## `interface_method`
+///
+/// ```test
+/// #[interface_method(id = 0)]
+/// ```
+///
+/// All the method in the trait with `exonum_interface` attribute should have `interface_method`
+/// attribute with unsigned integer value. All the method IDs should be unique.
 #[proc_macro_attribute]
 pub fn exonum_interface(attr: TokenStream, item: TokenStream) -> TokenStream {
     exonum_interface::impl_exonum_interface(attr, item)
 }
 
+/// Meta-information attribute for interface methods.
+///
+/// # Fields
+///
+/// ## `id` (required)
+///
+/// ```text
+/// #[interface_method(id = 0)]
+/// ```
+///
+/// Numeric identifier of the method. Should be unique for every method in the trait.
+#[proc_macro_attribute]
+pub fn interface_method(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    // We don't modify the input stream, since `interface_method` attribute only
```

Comment:
It could be possible to add a doc comment with the method ID mention, like

```rust
/// # Method identifier
///
/// This method has [method identifier](link to docs?) 0.
```